src/drive/web/modules/views/Folder/FolderViewBody.jsx

```diff
@@ -98,6 +101,7 @@ const FolderViewBody = ({
   const isSharingContextEmpty = Object.keys(sharingsValue).length <= 0
 
   const { syncingFakeFile } = useSyncingFakeFile({ isEmpty, queryResults })
+  const isEncFolder = isEncryptedFolder(client.store.getState())
```

Comment:
Why do we need to read it from the store? Don't we already have the `io.cozy.files` here? And then read its metadata without having to store it in the store? If you can avoid to manipulate / store in redux you should do.

Reply:
For now in the component we only have the dirID, but we could pass the whole `io.cozy.files` in props. But why is it better than reading the store? Is it a costful operation? Or is it because we want to less rely on the store?

Reply:
Having to read it from the store, makes the "externalization" harder. We tend to remove the use of redux in our app's land code. Why? Because of its verbosity and because we (mainly) duplicate the informations. Redux's store should only be used to store DATA from your cozy, not for the app's state. And since the communication between the app and your "cozy" is done by CozyClient, only cozyclient should store data in redux. Elsewhere, context should be used. 

Here, I think the UI can handle that directly by reading if the current folder is encrypted or not. Like that, no need to dispatch action to set the attribute or else. Only one path to read the info, and one path to set it.

Reply:
Ok, I changed the behaviour to directly read from the `displayedFolder` props